entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/config/EntityUpdateJobFactory.java

```diff
@@ -0,0 +1,171 @@
+        this.stepBuilderFactory = stepBuilderFactory;
+        this.itemListener = itemListener;
+        this.scheduledTaskService = scheduledTaskService;
+    }
+
+    // TODO this is a execution, we should seperate it from the factory methods and step creations
+    /**
+     * Job for updating a single entity. Expects `entityId` string in JobParameters. This would
+     * typically be run synchronously
+     */
+    public Job updateSingleEntity() {
+        return this.jobBuilderFactory
+                .get(JOB_UPDATE_SINGLE_ENTITY)
+                .incrementer(new RunIdIncrementer())
+                // this job is always launched from web requests, so synchronousTaskExecutor is used. It
+                // also directly retrieves entities from the EntityRecord database.
+                .start(synchronousUpdate())
+                .build();
+    }
+
+    /**
+     * Creates a step for synchronous update
+     * @return
+     */
+    private Step synchronousUpdate() {
+        return this.stepBuilderFactory
+                .get(STEP_UPDATE_ENTITY)
+                .<BatchEntityRecord, BatchEntityRecord>chunk(1)
+                .reader(getReader(true))
+                .processor((ItemProcessor<BatchEntityRecord, BatchEntityRecord>) applicationContext.getBean(compositeProcessor))
+                .writer((ItemWriter<BatchEntityRecord>) applicationContext.getBean(compositeEntityInsertionWriter))
+                .listener((ItemProcessListener<? super BatchEntityRecord, ? super BatchEntityRecord>) itemListener)
+                .faultTolerant()
+                .skipPolicy(noopSkipPolicy)
+                .taskExecutor((TaskExecutor)applicationContext.getBean(WEB_REQUEST_JOB_EXECUTOR))
+                .throttleLimit(emConfig.getBatchUpdatesThrottleLimit())
+                .listener(stepExecutionListener(List.of(ScheduledUpdateType.FULL_UPDATE), true))
+                .build();
+
+    }
+
+
+    /**
+     * Creates Item reader based on synchronus and async updates
+     * @param synchronous
+     * @return
+     */
+    private ItemReader<BatchEntityRecord> getReader(boolean synchronous) {
+        return synchronous ? (EntityRecordDatabaseReader) applicationContext.getBean(SINGLE_ENTITY_RECORD_READER)
+                : (SynchronizedItemStreamReader)applicationContext.getBean(SCHEDULED_TASK_READER);
+    }
+
+
+    /**
+     * Creating it as a bean as the processor list is same for all the Internal Task of EM.
+     * For performnace will access them from application context than creating a list for every request
+     * @see <a href="http://docs.google.com/document/d/16k9PcCMFwl2LXjnnzotZRPc-QqM-Ar1D0VELHt4t_hA/edit?tab=t.0#heading=h.fj6e15rbq64q"></a> }
+     *
+     * Creates the processor list -
+     *    Processors: Dereference + consolidation + metrics + validation
+     * @return
+     */
+    @Bean(compositeProcessor)
+    private ItemProcessor<BatchEntityRecord, BatchEntityRecord> compositeUpdateProcessor() {
```

Comment:
we need to refactor this code, the factory should not register beans, it should only create the objects. The method should get the list with the names of the processor. We should follow spring boot design and create need to create an EntityManagementAutoConfig which instantiates the required beans using the factory. Also, the bean names should be changed to indicate their functionality .. compositeProcessor is to generic. We should have something like "entityUpdateItemProcessors" as bean name